clcache.py

```diff
@@ -164,33 +164,43 @@ def clean(self, stats, maximumSize):
         except ImportError:
             walker = os.walk
 
-        objects = [os.path.join(root, "object")
-                   for root, folder, files in walker(self.objectsDir)
-                   if "object" in files]
+        objectInfos = []
+        for x in xrange(16):
+            try:
+                objects = [os.path.join(root, "object")
+                       for root, folder, files in walker(self.objectsDir)
+                       if "object" in files]
+
+                objectInfos = [(os.stat(fn), fn) for fn in objects]
```

Comment:
> it is mainly that users can delete the cache without causing builds to fail.

I don't think we should support that because when we start caring here, we must handle all cases where some random outside process does anything with the cache. Especially since we have `clcache -C` now, there should not be a reason to manually touch the cache while build processes are running.